src/utils/ucc_component.c

```diff
@@ -108,8 +101,11 @@ ucc_status_t ucc_components_load(const char *framework_name,
         if (n_loaded == ifaces_array_size) {
             ifaces_array_size += 8;
             ifaces = (ucc_component_iface_t **)ucc_realloc(
```

Comment:
I think it would be better to allocate once, after we know how many we need.

Reply:
i will allocate large enough storage (globbuf.gl_pathc) - it will be larger than required. Then i will shrink it to the actual size with ucc_realloc. I can't compute the size in advance, since in order to do that we need to try loading those components first (and store the result somewhere).